src/main/java/frc/robot/subsystems/drive/DriveSubsystem.java

```diff
@@ -452,6 +474,65 @@ public void addVisionMeasurement(Pose2d pose, double timestampSeconds,
                 poseEstimator.addVisionMeasurement(pose, timestampSeconds, visionMeasurementStdDevs);
         }
 
+        // this method has a LOT of suppliers - so short explanation of why they're good
+        // basically, when a button binding is called in robotContainer, it makes an
+        // object
+        // represnting the command. this object does not change, and it does not
+        // actually look
+        // at the command each time - it just calls the one that was constructed at
+        // robotInit
+        // by using a supplier, the robot knows 'hey, this value might change, so i
+        // should
+        // check it every time i use this object' thus allowing it to change
+        public Command driveToPoseCommand(Supplier<RobotPosition> robotPosition) {
+                return this.run(() -> {
+                        // basically, bad things can happen if we try to update a normal boolean within
+                        // a lambda and access it outside that lambda, but atomic booleans prevent these
+                        // risks
+                        atGoal.set(false);
+
+                        if (DriverStation.getAlliance().isPresent() && DriverStation.getAlliance().get() == Alliance.Blue) {
+                                isBlueAlliance = () -> true;
+                        } else {
+                                isBlueAlliance = () -> false;
+                        }
```

Comment:
```suggestion
                        isBlueAlliance = () -> DriverStation.getAlliance().isPresent() && DriverStation.getAlliance().get() == Alliance.Blue;
```

nit :pick: , but these are equivalent. Leave as is if you feel the current formulation is more readable